tests/Unit/Evolution/Systems/GeneralizedHarmonic/Test_Constraints.cpp

```diff
@@ -748,6 +748,52 @@ void test_constraint_energy_analytic(const Solution& solution,
       numerical_approx);
 }
 
+// Test the return-by-value normalized constraint energy function using random
+// values
+template <typename DataType, size_t SpatialDim, typename Frame>
+void test_constraint_energy_normalization_random(
+    const DataType& used_for_size) {
+  pypp::check_with_random_values<1>(
+      static_cast<Scalar<DataType> (*)(
+          const tnsr::iaa<DataType, SpatialDim, Frame>&,
+          const tnsr::iaa<DataType, SpatialDim, Frame>&,
+          const tnsr::ijaa<DataType, SpatialDim, Frame>&,
+          const tnsr::II<DataType, SpatialDim, Frame>&, const Scalar<DataType>&,
+          double)>(
+          &gh::constraint_energy_normalization<DataType, SpatialDim, Frame>),
+      "TestFunctions", "constraint_energy_normalization", {{{-1.0, 1.0}}},
+      used_for_size);
+}
+
+// Test the return-by-value normalized constraint energy function for Minkowski
+void test_constraint_energy_normalization_minkowski(
```

Comment:
I don't think this test is needed since the random values will cover it. Also, I'm not sure if the dimensional constant is defined as you go to flat space.

Reply:
Don't drop known-special-case tests just because there are python tests.  The python tests rely on there not being the same bug in the C++ and python implementations, which we have no way to verify (and which has happened).

(I haven't followed this PR and don't know whether this specific test makes sense.)

Reply:
Yes I agree keeping this specific test is a good idea.